packages/react-client/src/trackManager.ts

```diff
@@ -1,111 +1,141 @@
+import { useEffect, useState } from "react";
 
-export class TrackManager<PeerMetadata, TrackMetadata> implements GenericTrackManager<TrackMetadata> {
-  private readonly mediaManager: GenericMediaManager;
-  private readonly tsClient: FishjamClient<PeerMetadata, TrackMetadata>;
-  private currentTrackId: string | null = null;
+interface TrackManagerConfig<PeerMetadata, TrackMetadata> {
+  mediaManager: MediaManager;
+  tsClient: FishjamClient<PeerMetadata, TrackMetadata>;
+}
 
-  constructor(tsClient: FishjamClient<PeerMetadata, TrackMetadata>, deviceManager: GenericMediaManager) {
-    this.mediaManager = deviceManager;
-    this.tsClient = tsClient;
-  }
+export const useTrackManager = <PeerMetadata, TrackMetadata>({
+  mediaManager,
+  tsClient,
+}: TrackManagerConfig<PeerMetadata, TrackMetadata>): TrackManager<TrackMetadata> => {
+  const [currentTrackId, setCurrentTrackId] = useState<string | null>(null);
+  const [currentTrackMiddleware, setCurrentTrackMiddleware] = useState<TrackMiddleware>(null);
+
+  useEffect(() => {
+    const disconnectedHandler = () => {
+      setCurrentTrackId(null);
+    };
 
-  private getPreviousTrack = (): Track<TrackMetadata> => {
-    if (!this.currentTrackId) throw Error("There is no current track id");
+    tsClient.on("disconnected", disconnectedHandler);
 
-    const prevTrack = getRemoteOrLocalTrack<PeerMetadata, TrackMetadata>(this.tsClient, this.currentTrackId);
+    return () => {
+      tsClient.off("disconnected", disconnectedHandler);
+    };
+  }, [tsClient]);
+
+  function getPreviousTrack(): Track<TrackMetadata> {
+    if (!currentTrackId) throw Error("There is no current track id");
+
+    const prevTrack = getRemoteOrLocalTrack<PeerMetadata, TrackMetadata>(tsClient, currentTrackId);
 
     if (!prevTrack) throw Error("There is no previous track");
 
     return prevTrack;
-  };
+  }
 
-  public getCurrentTrack = (): Track<TrackMetadata> | null => {
-    return getRemoteOrLocalTrack<PeerMetadata, TrackMetadata>(this.tsClient, this.currentTrackId);
-  };
+  async function setTrackMiddleware(middleware: TrackMiddleware): Promise<void> {
+    const currentTrack = getCurrentTrack();
+    const mediaTrack = mediaManager.getTracks()[0];
 
-  public initialize = async (deviceId?: string) => {
-    this.mediaManager?.start(deviceId ?? true);
-  };
+    if (!currentTrack || !mediaTrack) return;
 
-  public stop = async () => {
-    this?.mediaManager?.stop();
-  };
+    const trackToSet = middleware ? middleware(mediaTrack) : mediaTrack;
+    await tsClient.replaceTrack(currentTrack.trackId, trackToSet);
 
-  public cleanup = () => {
-    this.currentTrackId = null;
-  };
+    setCurrentTrackMiddleware(() => middleware);
+  }
 
-  public startStreaming = async (
+  function getCurrentTrack(): Track<TrackMetadata> | null {
+    return getRemoteOrLocalTrack<PeerMetadata, TrackMetadata>(tsClient, currentTrackId);
+  }
+
+  async function initialize(deviceId?: string) {
```

Comment:
nit: In the future, we need to decide on a consistent code style for functions (fat arrow or function keyword). Can we refactor this file to use only one style?

Reply:
converted to `function`